.github/workflows/build_and_test.yml

```diff
@@ -39,18 +39,15 @@ jobs:
       # TODO: investigate how to run pre-commit with `venv`
       - uses: pre-commit/action@2c7b3805fd2a0fd8c1884dcaebf91fc102a13ecd # v3.0.1
 
-      - name: CMake format
-        run: |
-          ./ci/cmake-format-all.sh
-          git diff --exit-code
+      - name: Fetch dev branch
+        run: git fetch origin dev
```

Comment:
Maybe we want to run it as part of `clang-tidy-all.sh`?

Reply:
fixed